TESTING.md

```diff
@@ -47,6 +47,17 @@ Bugs fixes should include a unit test case which exercises the bug.
 A bug fix may also include new assertions in an existing integration tests for the
 API endpoint.
 
+### Integration tests guidelines
```

Comment:
I'm not sure if "Integration tests guidelines" is the right section header here. The entire doc is sort of a guide for tests and has references to integration testing littered all over. This section specifically talks about the 'environment for integration testing'. So it might be better to change the section name to say that. Also, to what effect are we interested in doc-ing the environment? One thing that came to mind is whether the test can be run in parallel with other integration tests or not and that's an environment consideration as well.